src/qaqc/01-organize_raw_data.Rmd

```diff
@@ -20,83 +20,96 @@ all_data <- list.files(path = 'data/api/', full.names = TRUE) %>%
-      select(-c(ph_7_nice, ph_7_other))
-  }
-
-  cal_table <- bind_rows(cal_table) %>%
-    distinct(.keep_all = TRUE) %>%
-    group_by(site) %>%
-    mutate(DT = round_date(DT, "15 minutes")) %>%
-    padr::pad(by = 'DT', group = 'site')
+cal_files <- list.files("data/calibration_reports", pattern=".html")
+
+cal_tabler <- function(cal_files){
+  
+  cal <- rvest::read_html(paste0("data/calibration_reports/", cal_files)) %>%
+    rvest::html_nodes('div') %>%
+    rvest::html_text() %>%
+    as_tibble()
+  
+  rdo <- cal %>% filter(grepl("RDO", value)) %>% pull() %>% str_replace_all(., " ", "")
+  
+  ph_orp <- cal %>% filter(grepl("pH/ORP", value)) %>% pull() %>% str_replace_all(., " ", "")
+  
+  conductivity <- cal %>% filter(grepl("Conductivity",value)) %>% pull() %>% str_replace_all(., " ", "")
+  
+  turbidity <- cal %>% filter(grepl("Turbidity",value)) %>% pull() %>% str_replace_all(., " ", "")
+  
+  time_mst <- paste0(str_sub(cal_files, -13, -12),":", str_sub(cal_files, -11, -10))
+  
+  date <- paste0(str_sub(cal_files, -22, -19),"-", str_sub(cal_files, -18, -17),"-", str_sub(cal_files, -16, -15))
+  
+  cal_table <- tibble(site = sub("\\_.*", "", cal_files),
+                      
+                      DT = ymd_hm(paste0(date, " ", time_mst)),
+                      
+                      #Dissolved Oxygen
+                      rdo_slope = str_match(rdo, "Slope\\s*(.*?)\\s*Offset")[,2],
+                      rdo_offset = str_match(rdo, "Offset\\s*(.*?)\\s*mg/L")[,2],
+                      rdo_100 = str_match(rdo, "PreMeasurement\\s*(.*?)\\s*%SatPost")[,2],
+                      rdo_conc = str_match(rdo, "Concentration\\s*(.*?)\\s*mg/LPreMeasurement")[,2],
+                      rdo_temp = str_match(rdo, "Temperature\\s*(.*?)\\s*°C")[,2],
+                      rdo_pressure = str_match(rdo, "Pressure\\s*(.*?)\\s*mbar")[,2],
+                      
+                      #pH
+                      ph_slope_pre = str_match(ph_orp, "Offset1Slope\\s*(.*?)\\s*mV/pH")[,2],
+                      ph_offset_pre = str_match(ph_orp, "mV/pHOffset\\s*(.*?)\\s*mVSlopeandOffset2")[,2],
+                      ph_slope_post = str_match(ph_orp, "Offset2Slope\\s*(.*?)\\s*mV/pH")[,2],
+                      ph_offset_post = str_match(ph_orp, paste0(ph_slope_post,"mV/pHOffset\\s*(.*?)\\s*mVORPORP"))[,2],
+                      ph_7_nice = str_sub(str_match(ph_orp, "PostMeasurementpH7\\s*(.*?)\\s*mVCal")[,2], 10, nchar(str_match(ph_orp, "PostMeasurementpH7\\s*(.*?)\\s*mVCal")[,2])),
+                      ph_7_other = str_sub(str_match(ph_orp, "PostMeasurementpH6\\s*(.*?)\\s*mVCal")[,2], 10, nchar(str_match(ph_orp, "PostMeasurementpH6\\s*(.*?)\\s*mVCal")[,2])),
+                      ph_7 = ifelse(is.na(ph_7_nice), ph_7_other, ph_7_nice),
+                      
+                      #ORP
+                      orp_offset = ifelse(is.na(str_match(ph_orp, "Zobell'sOffset\\s*(.*?)\\s*mVTemperature")[,2]),
```

Comment:
You could consider coercing the string to_lowercase() or to_uppercase() so that you don't need an ifelse. I'm going to guess that in some future, this will not be the only variable that has a mix of spelling/case, so it could be helpful longterm.